src/main/java/org/sonar/plugins/github/MarkDownUtils.java

```diff
@@ -22,16 +22,27 @@
 import java.io.UnsupportedEncodingException;
 import java.net.URLEncoder;
 import javax.annotation.Nullable;
+import org.sonar.api.BatchComponent;
+import org.sonar.api.CoreProperties;
+import org.sonar.api.batch.InstantiationStrategy;
+import org.sonar.api.config.Settings;
 
-public class MarkDownUtils {
+@InstantiationStrategy(InstantiationStrategy.PER_BATCH)
+public class MarkDownUtils implements BatchComponent {
 
   private static final String IMAGES_ROOT_URL = "https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/";
+  private final String baseUrl;
 
-  private MarkDownUtils() {
-    // Utility class
+  public MarkDownUtils(Settings settings) {
+    // If server base URL was not configured in SQ server then is is better to take URL configured on batch side
+    String baseUrl = settings.hasKey(CoreProperties.SERVER_BASE_URL) ? settings.getString(CoreProperties.SERVER_BASE_URL) : settings.getString("sonar.host.url");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "baseUrl" which hides the field declared at line 34. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)